Update Enketo to version 7.3.1

RuthTurk opened the pull request:
Adapts to the new Enketo monorepo, Yarn-based build system, and Node.js v20 upgrade.

Requires:
* https://github.com/kobotoolbox/enketo-literacy-test-widget/pull/42
* https://github.com/kobotoolbox/enketo-image-customization-widget/pull/7

RuthTurk committed:
update dockerfile to be compatabile with enketo update (v7.2.0) and fix updated paths in disclaimer-css.patch

RuthTurk committed:
uncomment commands to add custom widgets

RuthShryock commented:
https://github.com/kobotoolbox/enketo-literacy-test-widget/pull/42/commits/8ee65b7ff74f6c2502c86ecd0a1bb8ab3f9670a6

RuthShryock commented:
https://github.com/kobotoolbox/enketo-image-customization-widget/pull/7/commits/15a1096e2e924cba81caedd69aa17c5920823e26

RuthTurk committed:
update widget links to pull commit with enketo-core update

jnm commented on Dockerfile:
```diff
@@ -21,19 +33,21 @@ COPY config-at-build-time.json config/config.json
-    npx grunt && \
-    npm prune --production && \
-    rm config/config.json
+RUN yarn install --frozen-lockfile \
+    && yarn cache clean \
+    && yarn workspace enketo-express add https://github.com/kobotoolbox/enketo-image-customization-widget#15a1096e2e924cba81caedd69aa17c5920823e26 -W \
+    && yarn workspace enketo-express add https://github.com/kobotoolbox/enketo-literacy-test-widget#8ee65b7ff74f6c2502c86ecd0a1bb8ab3f9670a6 -W \
+    && yarn workspace enketo-express run build \
+    && rm config/config.json
```
I'm getting this complaint trying to build locally:
```
error enketo@: The engine "yarn" is incompatible with this module. Expected version ">= 1.22.22 <2". Got "1.22.19"
```

Also, the `cache clean` step should come at the end, after we've done our last `yarn` command. It's job is to avoid making the final Docker image larger than it has to be, but if we clean the cache and continue to use yarn, we'll refill the cache again

RuthTurk committed:
update Dockerfile to use enketo version 7.3.1

jnm committed:
Merge remote-tracking branch 'origin/master' into update-enketo-7.2.0

jnm committed:
Simplify Dockerfile and use pinned Enketo version

Avoid checking out the tip of the default branch of the Enketo
repository at build time, and instead adhere to the version used in the
base image
